#Probability

P(E) =  The probability of event E

* E and F are *dependent* if knowing something about whether E happens gives us information about whether F happens. Otherwise they are *independent*


* Mathematically, we say that two events E and F are independent if the probability that they both happen is the product of the probabilities that each one happens:

    * P(E, F) =  P(E) * P(F)

###Conditional Probability

* If two events E and F are independent, then by definition we have:

    * P(E,F) = P(E) * P(F)

* If they are not necessarily independent (and if the probability of F is not zero), then we define the probability of E "conditional on F" as:
    * P(E|F) = P(E, F)/P(F) 

* You should think of this as the probability that E happens, given that we know that F happens, we often rewrite as:
    * P(E, F) = P(E|F) * P(F)

* When E and F are independent, you can check that this gives:
    * P(E|F) = P(E)
    * Which is the mathematical way of expressing that knowing F occurred gives us no additinal information about whether E occurred

* Here is a common example
    * A family with two unkown children
        * Each child is equally likely to be a boy or girl
        * The gender of the second child is independent of the gender of the first child

* Now we can begin to ask what is the probability of an event
    * Both children are girls (B) 1/4
    * The older child is a girl (G) 1/2
    * Using the definition of conditional probability:
        * P(B|G) = P(B,G)/P(G) = P(B)/P(G) = 1/4 / 1/2 = 1/2
        * P(B,G) is just P(B) intuitively

* Lets do another
    * Both children are girls (B) 1/4
    * At least one of the children is a girl (L) 3/4 -- P(at least one) = 1 - P(not at all)
        * P(B|L) = P(B,L)/P(L) = P(B)/P(L) =  1/4 / 3/4 = 1/3

* We can check this by "generating" a lot of families

In [3]:
import random

In [4]:
def random_kid():
    return random.choice(['boy', 'girl'])

both_girls = 0
older_girl = 0
either_girl = 0

random.seed(0)
for _ in range(10000):
    younger = random_kid()
    older = random_kid()
    if older == 'girl':
        older_girl += 1
    if older == 'girl' and younger == 'girl':
        both_girls += 1
    if older == 'girl' or younger == 'girl':
        either_girl += 1

print("P(both | older): ", both_girls / older_girl)
print("P(both | either): ", both_girls / either_girl)

P(both | older):  0.5007089325501317
P(both | either):  0.3311897106109325


* P(both | older):  0.5007089325501317 = 1/2
* P(both | either):  0.3311897106109325 = 1/3

* This proves the theory

###Bayes's Theorem